Main.hs

```diff
@@ -50,38 +54,38 @@ basilica origin db emailChan = scottyApp $ do
     withPost json =<< param "id"
   get "/posts" $ do
     since <- maybeParam "after"
-    liftIO (getPostsSince db since) >>= json
+    liftDB db (getPostsSince since) >>= json
   post "/posts" $ with400 $ postPost Nothing
   post "/posts/:id" $
     with400 $ (Just <$> param "id") >>= postPost
   post "/codes" $
     with400 $ do
       emailAddress <- param "email"
-      liftIO $ do
-        code <- createCode db emailAddress
-        maybe (return ()) (writeChan emailChan . (emailAddress, )) code
+      liftDB db $ do
+        code <- createCode emailAddress
+        maybe (return ()) (liftIO . writeChan emailChan . (emailAddress,)) code
       status status200
   post "/tokens" $
     with400 $ do
       code <- param "code"
-      token <- liftIO (createToken db code)
-      maybe (status status401) (\t -> liftIO (withUser db t) >>= json) token
+      token <- liftDB db (createToken code)
+      maybe (status status401) (\t -> liftDB db (withUser t) >>= json) token
   post "/users" $
     with400 $ do
       email <- param "email"
       name <- param "name"
       if isValidName name then do
-        user <- liftIO (createUser db email name)
-        when (isJust user) $ liftIO $ do
-           code <- fromJust <$> createCode db email
-           writeChan emailChan (email, code)
+        user <- liftDB db (createUser email name)
+        when (isJust user) $ liftDB db $ do
+           code <- fromJust <$> createCode email
+           liftIO $ writeChan emailChan (email, code)
```

Comment:
Any `DatabaseM` things still need to be `liftIO`'d, even within a `liftDB` block.